### ISSUE TITLE
CombinableX509TrustManager can't completed all trustManager's check

### ISSUE THREAD

**Issue by manbucy** [Hakky54/sslcontext-kickstart#181]
**Describe the bug**
Hi, i found that CombinableX509TrustManager can't completed all trustManager's check, when some trustManager's trustdCerts is empty.

**To Reproduce**
```java
public class ElasticsearchTest {
    @Test
    public void test_connect_elasticsearch_server() throws IOException {
        SSLFactory sslFactory = SSLFactory.builder()
                .withSystemTrustMaterial()
                .withTrustMaterial(CertificateUtils.loadCertificate("http_ca.crt"))
                .build();
        BasicCredentialsProvider basicCredentialsProvider = new BasicCredentialsProvider();
        basicCredentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials("elastic", "es123456"));

        RestClient restClient = RestClient.builder(new HttpHost("192.168.169.2", 39200, "https"))
                .setHttpClientConfigCallback(httpAsyncClientBuilder -> httpAsyncClientBuilder
                        .setDefaultCredentialsProvider(basicCredentialsProvider)
                        .setSSLContext(sslFactory.getSslContext())
                        .setSSLHostnameVerifier(sslFactory.getHostnameVerifier())
                )
                .build();
        RestClientTransport restClientTransport = new RestClientTransport(restClient, new JacksonJsonpMapper());
        ElasticsearchClient elasticsearchClient = new ElasticsearchClient(restClientTransport);
        IndicesResponse indices = elasticsearchClient.cat().indices();
    }
}
```
in withSystemTrustMaterial(), it would load Windows-ROOT keystore and Windows-MY keystore, but now my Windows-MY keystore certicate is empty.
in withTrustMaterial(CertificateUtils.loadCertificate("http_ca.crt")), it would load the elasticsearch's certicate.
![trustmanagers](https://user-images.githubusercontent.com/24501621/167653717-e786ea91-e646-4e20-b1ac-9d96966ec06c.png)

the error is:
```
Caused by: java.security.InvalidAlgorithmParameterException: the trustAnchors parameter must be non-empty
	at java.security.cert.PKIXParameters.setTrustAnchors(PKIXParameters.java:200)
	at java.security.cert.PKIXParameters.<init>(PKIXParameters.java:120)
	at java.security.cert.PKIXBuilderParameters.<init>(PKIXBuilderParameters.java:104)
	at sun.security.validator.PKIXValidator.<init>(PKIXValidator.java:99)
	at sun.security.validator.Validator.getInstance(Validator.java:181)
	at sun.security.ssl.X509TrustManagerImpl.getValidator(X509TrustManagerImpl.java:299)
	at sun.security.ssl.X509TrustManagerImpl.checkTrustedInit(X509TrustManagerImpl.java:175)
	at sun.security.ssl.X509TrustManagerImpl.checkTrusted(X509TrustManagerImpl.java:245)
	at sun.security.ssl.X509TrustManagerImpl.checkServerTrusted(X509TrustManagerImpl.java:140)
	at nl.altindag.ssl.trustmanager.CompositeX509ExtendedTrustManager.lambda$checkServerTrusted$6(CompositeX509ExtendedTrustManager.java:110)
	at nl.altindag.ssl.trustmanager.CombinableX509TrustManager.checkTrusted(CombinableX509TrustManager.java:40)
	at nl.altindag.ssl.trustmanager.CompositeX509ExtendedTrustManager.checkServerTrusted(CompositeX509ExtendedTrustManager.java:110)
	at sun.security.ssl.CertificateMessage$T12CertificateConsumer.checkServerCerts(CertificateMessage.java:630)
	at sun.security.ssl.CertificateMessage$T12CertificateConsumer.onCertificate(CertificateMessage.java:471)
	at sun.security.ssl.CertificateMessage$T12CertificateConsumer.consume(CertificateMessage.java:367)
	at sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:376)
	at sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:479)
	at sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:990)
	at sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:977)
	at java.security.AccessController.doPrivileged(Native Method)
	at sun.security.ssl.SSLEngineImpl$DelegatedTask.run(SSLEngineImpl.java:924)
	at org.apache.http.nio.reactor.ssl.SSLIOSession.doRunTask(SSLIOSession.java:288)
	at org.apache.http.nio.reactor.ssl.SSLIOSession.doHandshake(SSLIOSession.java:356)
	... 9 more
```

I found in CombinableX509TrustManager will check all the TrustManager whether to support the current certificate chain. As a result of my second TrustManage from `Windows-MY` trustCerts is empty, so when executing `callBackConsumer.checkTrusted(trustManager);` method throws `InvalidAlgorithmParameterException`, but my third TrustManager actually can trust the current certificate.
![image](https://user-images.githubusercontent.com/24501621/167658297-bf6c3dc8-b8be-4ce8-ae21-7980beca928b.png)

Whether all exceptions can be caught in `CombinableX509TrustManager.checkTrusted` so that all TrustManagers have a chance to be executed, or whether `callBackConsumer.checkTrusted(trustManager);` is not executed when the `trustCerts` of the `trustManager` is empty.

Although I could have avoided this problem in my example by removing `.withSystemTrustMaterial()`, I think it's still possible that someone could load multiple TrustManagers at the same time, and the trustCerts of one of them might be empty.

**Environmental Data:**
 - Java Version 1.8.0_321
 - sslcontext-kickstart Version 7.4.2
 - OS: Windows

**Comment by Hakky54** [comment]
Hi @manbucy 

Thank you for this detailed bug report. I validated it and this issue is indeed present. It seems like the trust manager is by default throwing a different exception when it tries to check if the counter party is trusted while having zero trust certificates. In your case this happens with one of the system keystore which does not have trusted certificates. I have applied a fix and also added a test. 

I will make a release right now so you can give it a try. Please let me know if this issue is still occurring with the latest version

**Comment by manbucy** [comment]
Hi @Hakky54 
This issue has been fixed in version 7.4.3, Thanks.
